apps/examples/stack_protection/stack_protection_main.c

```diff
@@ -0,0 +1,119 @@
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ * Author: Sangamanatha <sangam.swami@samsung.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <pthread.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+#define STACK_THREAD_SIZE			512
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+static pthread_addr_t stack_tc_thread()
+{
+	char ch;
+	int i = 0;
+	volatile char array[STACK_THREAD_SIZE];
+
+	while (1) {
+		printf("\nPress N - Test Stack access within range\n");
+		printf("Press O - Test Stack access in overflow\n");
+		ch = getchar();
+		if (ch == 'N' || ch == 'n') {
+			for (i = STACK_THREAD_SIZE / 2; i <= STACK_THREAD_SIZE; i++) {
+				array[i] = 'a';
+			}
+			printf(" Test stack access within range - Passed\n");
+		} else if (ch == 'O' || ch == 'o') {
+			for (i = 0; i < STACK_THREAD_SIZE; i++) {
+				array[i] = 'b';
+			}
+			printf(" Test stack access in overflow - Failed\n");
+		}
+	}
+
+	return 0;
+}
+
+static pthread_t create_stack_tc_thread(void)
+{
+	pthread_t tid;
+	pthread_attr_t attr;
+	struct sched_param sparam;
+	int ret = OK;
+
+	ret = pthread_attr_init(&attr);
+	if (ret != 0) {
+		fprintf(stderr, "failed to set pthread init(%d)\n", ret);
+		return -ret;
+	}
+
+	ret = pthread_attr_setschedpolicy(&attr, SCHED_RR);
+	if (ret != 0) {
+		fprintf(stderr, "failed to set policy(%d)\n", ret);
+		return -ret;
+	}
+
+	ret = pthread_attr_setstacksize(&attr, STACK_THREAD_SIZE);
+	if (ret != 0) {
+		fprintf(stderr, "failed to set stack size(%d)\n", ret);
+		return -ret;
+	}
+
+	ret = pthread_create(&tid, &attr, stack_tc_thread, NULL);
+	if (ret != 0) {
+		fprintf(stderr, "failed to create pthread(%d)\n", ret);
+		return -ret;
+	}
+
+	pthread_setname_np(tid, "stack tc thread");
+	pthread_join(tid, NULL);
+	return 0;
```

Comment:
I think you'd better return ret instead of 0.

Reply:
yes handled